### PR TITLE
Fix add basic auth to alert manager config

### PR DESCRIPTION
**Pull request by LucasRoesler**
## Description
**What**
- Update the alert manager config to send the basic auth credentials
when basic auth is enabled.
- This is required to support openfaas/faas#1049
- bump the gateway tag
- bump the chart version

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

This issue was raised by @alexellis on slack

## How Has This Been Tested?

Tested manually in Docker-for-mac

Pull the gateway image from openfaas/faas#1049
```sh
docker pull theaxer/gateway:0.10.0
```

Create `spammer.sh`
```sh
#! /bin/bash
gateway=localhost:31112
fnc=cows
sleepSec=0
while true; do
    echo "" | faas-cli invoke "$fnc" --gateway "$gateway"
    sleep $sleepSec
done
```

Deploy and test in k8s
```sh
kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml && \
kubectl -n openfaas create secret generic basic-auth \
    --from-literal=basic-auth-user=admin \
    --from-literal=basic-auth-password='testpass'
helm upgrade openfaas --install ./openfaas \
    --namespace openfaas  \
    --set basic_auth=true \
    --set gateway.image='theaxer/gateway:0.10.0' \
    --set faasnetesd.imagePullPolicy='IfNotPresent' \
    --set openfaasImagePullPolicy='IfNotPresent' \
    --set functionNamespace=openfaas-fn

export gateway=localhost:31112

# test that basic auth is required on the alert endpoint
curl -X POST http://$gateway/system/alert -d '{}'
# invalid credentials%

faas-cli -g $gateway login -u admin --password testpass
faas-cli -g $gateway store deploy cows

# in a new shell, run the spammer to test scaling
./spammer.sh

# watch the gateway logs for `Alert received.` message
kubectl -n openfaas logs $(kubectl -n openfaas get po -l "app=gateway" -o name) gateway -f
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by burtonr** [comment]
Looks good to me! I was able to verify the `/system/alert` endpoint is now using the basic auth, no longer accessible via curl:

```
$ curl -X POST http://127.0.0.1:8080/system/alert -d '{}'
invalid credentials
```
unfortunately, In testing this on KinD, it never hit the alert and ran out of time this evening to setup a new cluster to test this with. 

Looking at the change, and knowing that it works with the Swarm integration, I have confidence that this is work well done (as usual), although I suggest someone else do the double check, just to be sure!

**Comment by alexellis** [comment]
The cows function is funny, but it's not the quickest that we have. I created a new repo for testing -> https://github.com/alexellis/echo-fn - you may find it's easier to get this to auto-scale based upon QPS.

**Comment by bartsmykla** [comment]
@alexellis I'm on it

**Comment by bartsmykla** [comment]
So I've tested it using `minikube`, and everything worked as expected

File: chart/openfaas/values.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ faasnetesd:
     periodSeconds: 10
 
 gateway:
-  image: openfaas/gateway:0.9.14
+  image: openfaas/gateway:0.10.0
```
Comment (alexellis) [inline chart/openfaas/values.yaml]: Thanks for this PR. FYI your change went out in: https://github.com/openfaas/faas/releases/tag/0.10.2

Comment (alexellis) [reply chart/openfaas/values.yaml]: @LucasRoesler FYI

**Comment by LucasRoesler** [comment]
@alexellis  reverted the image version change

**Comment by alexellis** [comment]
Thank you for this fix.

You can now go ahead and find the latest gateway and bump in helm, and AMD64 YAML (not armhf/arm64)

**Pull request merged**